### ISSUE TITLE
Problem regarding loading a QAT SavedModel

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/model-optimization#644]
Prior to filing: check that this should be a bug instead of a feature request. Everything supported, including the compatible versions of TensorFlow, is listed in the overview page of each technique. For example, the overview page of quantization-aware training is [here](https://www.tensorflow.org/model_optimization/guide/quantization/training). An issue for anything not supported should be a feature request.

**Describe the bug**
Currently, I am using QAT inside a custom training loop context and it goes pretty smoothly. But the problem starts surfacing when I try to load the serialized SavedModel ([Colab Notebook](https://colab.research.google.com/gist/sayakpaul/b699db0949c13d7c9561f17edef9dc8f/scratchpad.ipynb)) - 

```
KeyError: '__inference_expanded_conv_depthwise_layer_call_and_return_conditional_losses_330835'
```

**System information**

I am using Colab with TensorFlow 2.4.1. 

**Describe the expected behavior**

The `SavedModel` should load as expected. 

**Describe the current behavior**

The `SavedModel` is not getting loaded as expected and the predictions of the QAT model are extremely spurious. 

**Code to reproduce the issue**
This zip file contains a Jupyter Notebook. Although it uses 4 GPUs to distribute the training, I believe if you disable the strategy scope it will run just fine on a Colab Notebook. 

[Knowledge_Distillation_MobileNetV2_QAT.ipynb.zip](https://github.com/tensorflow/model-optimization/files/6112489/Knowledge_Distillation_MobileNetV2_QAT.ipynb.zip)

**Screenshots**

**Additional context**

**Comment by daverim** [comment]
Seems that this issue is due to the student model not being saved correctly. Does the non-quantized version of the model have the same issue?

**Comment by sayakpaul** [comment]
No, it does not have the same issue. A more serious problem is that the predictions of the QAT model are spurious.

**Comment by daverim** [comment]
I'm having trouble recreating your issue, I tried with some random data in a simple test and it seems to work fine.
Just to check, your input is using the same preprocessing for the evaluation
```normalization_layer = experimental.preprocessing.Rescaling(1./127.5, offset=-1)```
?

**Comment by sayakpaul** [comment]
Yes, I am using the same preprocessing layer. 

Can you elaborate a bit more as to where the problem is coming from when you are trying to reproduce the issues?

**Comment by daverim** [comment]
I uploaded your notebook, for a simple test, I just quantize the get_student_model() directly and train one epoch with some generated data. While I can't test accuracy, it seems to load and save fine. I also tried with the distiller using the float model as the teacher and there was no issue with load and save (using just the CPU device)

I will try today with GPU and see if there is an issue caused by distributed training. Would you mind providing your teacher model and access to your bucket? (davidrim@google.com)

**Comment by sayakpaul** [comment]
@daverim thank you. The bucket is publicly accessible.

**Comment by daverim** [comment]
Hi was able to run training although only for one epoch. I did notice you are not using the student loss, and only the distillation loss. When you say the model is spurious, do you mean that the student model predictions are bad or that the student model predictions are not the same as during training?

**Comment by sayakpaul** [comment]
Yeah student loss is not required in our training. When I say they are spurious I mean they predict the same output always.

**Comment by teijeong** [comment]
Ping @daverim  :)

**Comment by daverim** [comment]
Sorry, this fell off my radar. My trained model does not have the same outputs for every inputs. Did you try different learning rates? The quantized student will have very different inputs. Also, upgrade to tf-2.5 to see if your problem persists.

**Comment by sayakpaul** [comment]
I will keep this issue open until then. 

But I would suggest training this for at least 10 epochs on your end to verify the issue.
